### DISCUSSION TITLE
Wallet disconnect doesn't work

### DISCUSSION THREAD

**Discussion by npek** [rainbow-me/rainbowkit#902] [Q&A]
First of all, after I try to connect my wallet, the MetaMask icon sits pending forever until I exit the modal.
Then, after my wallet is connected, if I try to click the dropdown and disconnect, nothing happens. My wallet remains connected and the dropdown actually becomes completely unresponsive. I have to manually disconnect my wallet in MetaMask itself, then I have to refresh the page or click the chains button for the UI to correctly show that my wallet is no longer connected.
Am I implementing something wrong? Or is this a bug with RainbowKit itself?

**Comment by aalmada** [comment]
https://github.com/MetaMask/metamask-extension/issues/10353#issuecomment-1024690417
